palo_alto_cortex_xdr/icon_palo_alto_cortex_xdr/util/api.py

```diff
@@ -0,0 +1,301 @@
+                if len(all_incidents) >= total_count or total_count < 0:
+                    done = True
+
+                # Update the indices of search_from and search_to to we can request the next page
+                search_from = search_from + batch_size
+                search_to = search_to + batch_size if search_to + batch_size < total_count else total_count
+
+                # Add the updated page indices to the request body for when we request the next page
+                post_body["request_data"]["search_from"] = search_from
+                post_body["request_data"]["search_to"] = search_to
+            else:
+                done = True
+
+            # Back-off between making requests to the API.
+            time.sleep(1)
+
+        return all_incidents
+
+    def allow_or_block_file(self, file_hash, comment, incident_id=None, block_file=True):
+        if block_file:
+            endpoint = "/public_api/v1/hash_exceptions/blocklist/"
+        else:
+            endpoint = "/public_api/v1/hash_exceptions/allowlist/"
+
+        post_body = {
+            "request_data": {
+                "hash_list": [file_hash],
+                "comment": comment,
+            }
+        }
+
+        if incident_id:
+            post_body["request_data"]["incident_id"] = incident_id
+
+        block_action = "Block" if block_file else "Allow"
+
+        self.logger.info(f"Taking {block_action} action on file: {file_hash}")
+        self.logger.info(f"Endpoint: {endpoint}")
+
+        result = self._post_to_api(endpoint, post_body)
+
+        if result.get("reply"):  # reply will be true or false
+            self.logger.info(f"{block_action} action was successful")
+            return True
+        self.logger.warning(f"{block_action} action failed")
+        self.logger.warning(f"Result: {result}")
+        return False
+
+    ###########################
+    # Private Methods
+    ###########################
+    def _isolate_multiple_endpoints(self, endpoints, isolation_state):
+        if isolation_state:
+            api_endpoint = "/public_api/v1/endpoints/isolate/"
+        else:
+            api_endpoint = "/public_api/v1/endpoints/unisolate/"
+
+        endpoint_ids = []
+        for ep in endpoints:
+            endpoint_ids.append(ep.get("endpoint_id"))
+
+        self.logger.info(f"Isolation: {isolation_state}\nEndpoint IDs:{endpoint_ids}")
+        post_body = {
+            "request_data": {"filters": [{"field": "endpoint_id_list", "operator": "in", "value": endpoint_ids}]}
+        }
+
+        result = self._post_to_api(api_endpoint, post_body)
+        return result.get("reply")
+
+    def _isolate_endpoint(self, endpoint, isolation_state):
+        if isolation_state:
+            api_endpoint = "/public_api/v1/endpoints/isolate/"
+        else:
+            api_endpoint = "/public_api/v1/endpoints/unisolate/"
+
+        endpoint_id = endpoint[0].get("endpoint_id")
```

Comment:
The variable name is singular so I am assuming this is a string - why are you accessing an index on it?

Reply:
Renamed